ecllibrary/std/DataPatterns/Profile.ecl

```diff
@@ -514,6 +527,67 @@ EXPORT Profile(inFile,
         #UNIQUENAME(_MakeAttr);
         LOCAL %_MakeAttr%(STRING attr) := REGEXREPLACE('\\.', attr, '_');
 
+        // Determine if a UTF-8 string really contains UTF-8 characters
+        #UNIQUENAME(IsUTF8);
+        LOCAL BOOLEAN %IsUTF8%(UTF8 str) := EMBED(C++)
+            if (lenStr == 0)
+                return false;
+
+            const unsigned char*    bytes = reinterpret_cast<const unsigned char*>(str);
+            const unsigned char*    endPtr = bytes + lenStr;
```

Comment:
Technically you should have
size32_t sizeUtf8 = rtlUtf8Length(lenStr, str);
since length is the number of code points, not the number of bytes.  But that would be inefficient.

I think the clean way is a two loops.  One which scans all the ascii characters.  Then a check if (bytes < endPtr) of the last codepoint.

Reply:
Actually, I realized that this function shouldn't be using UTF8 as a parameter data type at all.  It should be DATA.  Making that change means lenStr makes sense again.